### PR TITLE
RavenDB-21996 IDocumentQueryCustomization documentation

### PR DESCRIPTION
**Pull request by Lwiel**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21996

### Additional description

Documentation update

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Documents/Session/IDocumentQueryCustomization.cs [hunk 1]
```diff
@@ -19,74 +19,97 @@ namespace Raven.Client.Documents.Session
...
+        /// <param name="action">Action with query result parameter. Defines the method that will be executed after query execution.</param>
+        /// <seealso ref="https://ravendb.net/docs/article-page/6.0/csharp/client-api/session/querying/how-to-customize-query#afterqueryexecuted"/>
         IDocumentQueryCustomization AfterQueryExecuted(Action<QueryResult> action);
 
         /// <summary>
-        ///     Callback to get the results of the stream
+        ///     Allows to access raw streaming query result (in form of BlittableJsonReaderObject) after the query execution.
         /// </summary>
+        /// <param name="action">Action with stream result parameter. Defines the method that will be executed after streaming query execution.</param>
+        /// <seealso ref="https://ravendb.net/docs/article-page/6.0/csharp/client-api/session/querying/how-to-customize-query#afterstreamexecuted"/>
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/IDocumentQueryCustomization.cs]: check with @maciejaszyk how are we linking

Comment (Lwiel) [reply src/Raven.Client/Documents/Session/IDocumentQueryCustomization.cs]: Fixed

File: src/Raven.Client/Documents/Session/IDocumentQueryCustomization.cs [hunk 2]
```diff
@@ -19,74 +19,93 @@ namespace Raven.Client.Documents.Session
...
         IDocumentQueryCustomization Timings(out QueryTimings timings);
         
         /// <summary>
-        ///   Instruct the query to wait for non stale results.
-        ///   This shouldn't be used outside of unit tests unless you are well aware of the implications
+        ///     Instruct the query to wait for non-stale results.
+        ///     This shouldn't be used outside of unit tests unless you are well aware of the implications.
         /// </summary>
-        /// <param name = "waitTimeout">Maximum time to wait for index query results to become non-stale before exception is thrown. Default: 15 seconds.</param>
+        /// <param name="waitTimeout">Maximum time in seconds to wait for index query results to become non-stale before exception is thrown. Default: 15 seconds.</param>
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Session/IDocumentQueryCustomization.cs]: Why "in seconds"? It's a TimeSpan

Comment (Lwiel) [reply src/Raven.Client/Documents/Session/IDocumentQueryCustomization.cs]: Fixed

**Pull request merged**